### PR TITLE
Modify staging release buildspec to support batch builds

### PR DESCRIPTION
**Pull request by abhay-krishna**
This PR modifies the staging release buildspec to support Codebuild batch builds.

/hold

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: release/staging-build.yml [hunk 1]
```diff
@@ -1,35 +1,245 @@
...
-  
-  post_build:
-    commands:
-    - export HOME="/home/imagebuilder"
-    - make release-ovas
-    
+batch:
+  fast-fail: true
+  build-graph:
+  - identifier: eks_anywhere_diagnostic_collector
```
Comment (abhay-krishna) [inline release/staging-build.yml:4]: Using underscores because identifier doesn't support dashes.

File: release/staging-build.yml [hunk 2]
```diff
@@ -1,35 +1,245 @@
...
+      variables:
+       PROJECT_PATH: projects/helm/helm
+       CLONE_URL: https://git-codecommit.us-west-2.amazonaws.com/v1/repos/helm.helm
+  - identifier: tink
+    buildspec: buildspec.yml
+    env:
+      variables:
+       PROJECT_PATH: projects/tinkerbell/tink
+       CLONE_URL: https://git-codecommit.us-west-2.amazonaws.com/v1/repos/tinkerbell.tink
+  - identifier: bottlerocket_bootstrap
```
Comment (jaxesn) [inline release/staging-build.yml:195]: This one depends on etcdadm at least, that may be it but worth a check

File: release/staging-build.yml [hunk 3]
```diff
@@ -1,35 +1,245 @@
...
+      variables:
+       PROJECT_PATH: projects/kubernetes-sigs/cluster-api
+       CLONE_URL: https://git-codecommit.us-west-2.amazonaws.com/v1/repos/kubernetes-sigs.vsphere-csi-driver
+  - identifier: cluster_api_provider_aws
+    buildspec: buildspec.yml
+    env:
+      variables:
+       PROJECT_PATH: projects/kubernetes-sigs/cluster-api-provider-aws
+       CLONE_URL: https://git-codecommit.us-west-2.amazonaws.com/v1/repos/kubernetes-sigs.cluster-api-provider-aws
+  - identifier: kind
```
Comment (jaxesn) [inline release/staging-build.yml:159]: This depends on etcdadm and cri-tools and local path provisioner, double check there aren't more.

**Review by jaxesn — CHANGES_REQUESTED** [review]
Double check through the repo for projects which use fetch artifacts, that should be a decent list of ones which have deps like this.

File: release/staging-build.yml (hunk 3 above)
Comment (abhay-krishna) [reply release/staging-build.yml:159]: yup i had them on before, but i think I removed them during testing. Will add them back.

**Comment by abhay-krishna** [comment]
I have included projecs in the buildspec that correspond to non-production-ready providers, like cloudstack and all of the Tinkerbell ones. Do we still need to build them? If not I can remove them.

**Review by jaxesn — APPROVED** [review]
/lgtm

**Pull request merged**